griptape/drivers/vector/peristable_local_vector_store_driver.py

```diff
@@ -0,0 +1,83 @@
+import dataclasses
+import json
+from pathlib import Path
+from typing import Union, Literal, cast, Dict
+from zipfile import ZipFile
+
+from attr import define, field
+
+from griptape.drivers import LocalVectorStoreDriver
+
+
+class EntryAwareJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if isinstance(o, PersistableLocalVectorStoreDriver.Entry):
+            d_dict = dataclasses.asdict(o)
+            d_dict["__Entry__"] = True
+            return d_dict
+        return super().default(o)
+
+
+def as_entry(dct: Dict):
+
+    if "__Entry__" in dct:
+        dct.pop("__Entry__")
+        return PersistableLocalVectorStoreDriver.Entry(**dct)
+
+    return dct
+
+
+@define
+class PersistableLocalVectorStoreDriver(LocalVectorStoreDriver):
+    """
+    Version of the LocalVectorStoreDriver that allows to serialize to a local file using either json or a zip 
+    depending on the suffix of the given file path.
+
+    :param file_path: `pathlib.Path` or path as string. Name and location of the save file. Extension must be zip or
+    json, if zip the generated file is compressed.
+    """
+
+    file_path: Union[Path, str] = field(kw_only=True, converter=Path)
+    file_is_zip: bool = field(init=False)
+
+    @file_path.validator
+    def _check_file_path(self, attribute, value):
+        if not value.suffix in (".json", ".zip"):
+            raise ValueError(
+                f"file must either be a json or a zip but file name is {self.file_path}"
+            )
+
+    def __attrs_post_init__(self):
+        self.file_is_zip = self.file_path.suffix == ".zip"
```

Comment:
Can we move this into a property? [HERE](https://github.com/griptape-ai/griptape/blob/main/griptape/drivers/sql/amazon_redshift_sql_driver.py#L18) is an example of how it can be implemented.